packages/iTwinUI-react/src/core/Table/actionHandlers/selectHandler.ts

```diff
@@ -56,6 +56,109 @@ export const onSelectHandler = <T extends Record<string, unknown>>(
   onSelect?.(selectedData, newState);
 };
 
+/**
+ * Handles selection when clicked on a row while shift is selected.
+ */
+export const onShiftSelectHandler = <T extends Record<string, unknown>>(
+  state: TableState<T>,
+  action: ActionType,
+  instance: TableInstance<T>,
```

Comment:
did you get a chance to look into whether `instance` should actually be optional or not? if it should be optional in the reducer, then it might make sense to also make it optional here

Reply:
To be safe, I have temporarily reverted it back to how it was. Since with Vilius's code in `vilius/shift-selection-demo`, `instance?` also worked. But I might be making more slight changes to this code. So I will confirm once again at the end on if everything works with instance being optional.